controllers/vaultpkisecret_controller.go

```diff
@@ -292,25 +309,26 @@ func (r *VaultPKISecretReconciler) handleDeletion(ctx context.Context, l logr.Lo
 	return nil
 }
 
-func (r *VaultPKISecretReconciler) addFinalizer(ctx context.Context, l logr.Logger, s *secretsv1beta1.VaultPKISecret) error {
+func (r *VaultPKISecretReconciler) addFinalizer(ctx context.Context, s *secretsv1beta1.VaultPKISecret) (bool, error) {
+	logger := log.FromContext(ctx).WithValues("finalizer", vaultPKIFinalizer)
 	if !controllerutil.ContainsFinalizer(s, vaultPKIFinalizer) {
 		controllerutil.AddFinalizer(s, vaultPKIFinalizer)
+		logger.V(consts.LogLevelDebug).Info("Adding finalizer")
 		if err := r.Client.Update(ctx, s); err != nil {
-			l.Error(err, "error updating VaultPKISecret resource")
-			return err
+			logger.Error(err, "Adding finalizer")
+			return false, err
 		}
+		return true, nil
+	} else {
+		logger.V(consts.LogLevelDebug).Info("Finalizer already added")
+		return false, nil
 	}
-
-	return nil
 }
 
 // SetupWithManager sets up the controller with the Manager.
 func (r *VaultPKISecretReconciler) SetupWithManager(mgr ctrl.Manager) error {
 	return ctrl.NewControllerManagedBy(mgr).
 		For(&secretsv1beta1.VaultPKISecret{}).
-		// Add metrics for create/update/delete of the resource
-		Watches(&secretsv1beta1.VaultPKISecret{},
-			&handler.InstrumentedEnqueueRequestForObject{}).
```

Comment:
Curious why this was removed? Are those metrics gathered elsewhere?

Reply:
It was causing the VDS CR to be reconciled twice.

Reply:
Sorry, I missed your second question. Yes we are still getting the same controller_* metrics for each controller:
e.g:  
```
# HELP controller_runtime_reconcile_errors_total Total number of reconciliation errors per controller
# TYPE controller_runtime_reconcile_errors_total counter
controller_runtime_reconcile_errors_total{controller="hcpauth"} 0
controller_runtime_reconcile_errors_total{controller="hcpvaultsecretsapp"} 0
controller_runtime_reconcile_errors_total{controller="vaultauth"} 0
controller_runtime_reconcile_errors_total{controller="vaultconnection"} 1
controller_runtime_reconcile_errors_total{controller="vaultdynamicsecret"} 0
controller_runtime_reconcile_errors_total{controller="vaultpkisecret"} 1
controller_runtime_reconcile_errors_total{controller="vaultstaticsecret"} 0
```

Reply:
> It was causing duplicate CR reconciliation. One for For() and one for Watches(). That's what was causing the failed status updates, and extra rotations.

Hmm, I'm not seeing any of that when testing a build off main with the PKI samples.

Either way I'm fine with removing it here, since it isn't present on any of the other controllers.

Reply:
If you have `-zap-log-level=5` set on the manager, you will see the duplicate enqueuing. Different reconcile IDs within quick succession.